magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/ActivityLifecycleAdapter.kt

```diff
@@ -2,46 +2,52 @@ package com.wealthfront.magellan.lifecycle
 
 import android.app.Activity
 import androidx.activity.ComponentActivity
-import androidx.annotation.IdRes
+import com.wealthfront.magellan.lifecycle.LifecycleOwner as MagellanLifecycleOwner
+import androidx.lifecycle.LifecycleOwner as ActivityLifecycleOwner
 import androidx.lifecycle.DefaultLifecycleObserver
 import com.wealthfront.magellan.core.Navigable
-import androidx.lifecycle.LifecycleOwner
+import com.wealthfront.magellan.R
 import com.wealthfront.magellan.ScreenContainer
 
 internal class ActivityLifecycleAdapter(
   private val navigable: Navigable,
-  private val context: Activity,
-  private val containerRes: Int
+  private val context: Activity
 ) : DefaultLifecycleObserver {
 
-  override fun onStart(owner: LifecycleOwner) {
+  override fun onCreate(owner: ActivityLifecycleOwner) {
+    if (navigable is MagellanLifecycleOwner && navigable.currentState == LifecycleState.Destroyed) {
```

Comment:
Makes sure that onCreate is only called after all the navigables are destroyed (application is destroyed)

Reply:
Probably an edge case we don't care about, but this won't call `onCreate` if the root navigable isn't a `LifecycleOwner`.

I can't think of a time where that would be the case though haha